integration/polkadot/incrementer.contract

```diff
@@ -0,0 +1 @@
":["[your_name] <[your_email]>"]},"image":null,"spec":{"constructors":[{"args":[{"label":"init_value","type":{"displayName":["i32"],"type":0}}],"default":false,"docs":[],"label":"new","payable":false,"returnType":{"displayName":["ink_primitives","ConstructorResult"],"type":2},"selector":"0x9bae9d5e"},{"args":[],"default":false,"docs":[],"label":"new_default","payable":false,"returnType":{"displayName":["ink_primitives","ConstructorResult"],"type":2},"selector":"0x61ef7e3e"}],"docs":[],"environment":{"accountId":{"displayName":["AccountId"],"type":6},"balance":{"displayName":["Balance"],"type":9},"blockNumber":{"displayName":["BlockNumber"],"type":12},"chainExtension":{"displayName":["ChainExtension"],"type":13},"hash":{"displayName":["Hash"],"type":10},"maxEventTopics":4,"staticBufferSize":16384,"timestamp":{"displayName":["Timestamp"],"type":11}},"events":[],"lang_error":{"displayName":["ink","LangError"],"type":4},"messages":[{"args":[{"label":"by","type":{"displayName":["i32"],"type":0}}],"default":false,"docs":[],"label":"inc","mutates":true,"payable":false,"returnType":{"displayName":["ink","MessageResult"],"type":2},"selector":"0x1d32619f"},{"args":[],"default":false,"docs":[],"label":"get","mutates":false,"payable":false,"returnType":{"displayName":["ink","MessageResult"],"type":5},"selector":"0x2f865bd9"}]},"storage":{"root":{"layout":{"struct":{"fields":[{"layout":{"leaf":{"key":"0x00000000","ty":0}},"name":"value"}],"name":"Incrementer"}},"root_key":"0x00000000","ty":1}},"types":[{"id":0,"type":{"def":{"primitive":"i32"}}},{"id":1,"type":{"def":{"composite":{"fields":[{"name":"value","type":0,"typeName":"<i32 as::ink::storage::traits::AutoStorableHint<::ink::storage\n::traits::ManualKey<3794263404u32, ()>,>>::Type"}]}},"path":["incrementer","incrementer","Incrementer"]}},{"id":2,"type":{"def":{"variant":{"variants":[{"fields":[{"type":3}],"index":0,"name":"Ok"},{"fields":[{"type":4}],"index":1,"name":"Err"}]}},"params":[{"name":"T","type":3},{"name":"E","type":4}],"path":["Result"]}},{"id":3,"type":{"def":{"tuple":[]}}},{"id":4,"type":{"def":{"variant":{"variants":[{"index":1,"name":"CouldNotReadInput"}]}},"path":["ink_primitives","LangError"]}},{"id":5,"type":{"def":{"variant":{"variants":[{"fields":[{"type":0}],"index":0,"name":"Ok"},{"fields":[{"type":4}],"index":1,"name":"Err"}]}},"params":[{"name":"T","type":0},{"name":"E","type":4}],"path":["Result"]}},{"id":6,"type":{"def":{"composite":{"fields":[{"type":7,"typeName":"[u8; 32]"}]}},"path":["ink_primitives","types","AccountId"]}},{"id":7,"type":{"def":{"array":{"len":32,"type":8}}}},{"id":8,"type":{"def":{"primitive":"u8"}}},{"id":9,"type":{"def":{"primitive":"u128"}}},{"id":10,"type":{"def":{"composite":{"fields":[{"type":7,"typeName":"[u8; 32]"}]}},"path":["ink_primitives","types","Hash"]}},{"id":11,"type":{"def":{"primitive":"u64"}}},{"id":12,"type":{"def":{"primitive":"u32"}}},{"id":13,"type":{"def":{"variant":{}},"path":["ink_env","types","NoChainExtension"]}}],"version":"4"}
```

Comment:
I tried to use `solang`-compiled contracts with different versions of `substrate-contracts-node`, but encountered incompatibilities. The errors included:
- With `solang-substrate-ci` ([latest version](https://github.com/hyperledger/solang-substrate-ci/pkgs/container/solang-substrate-ci/128945070?tag=75f66ac)):
```bash
Error uploading the code: Error decoding into dynamic value: Error at : Field deposit_held does not exist in our encoded data
```
- Deploying `solang`-compiled contracts with the latest version of `substrate-contracts-node` failed during instantiation because of differing metadata formats.
```bash
Failed to deserialize ink project metadata from contract metadata
```
- Even when attempting to use an older version of `substrate-contracts-node`, like v26, with `solang`-compiled contracts:
```bash
Error uploading the code: Error decoding into dynamic value: Error at : Field deposit_held does not exist in our encoded data
```
Currently, to test `solang-aqd`, I'm using `ink!`-compiled contracts with the most recent version of `substrate-contracts-node`. I'm using this approach until `solang-substrate-ci` gets updated. 
The goal is to use `solang`-compiled contracts once this update is implemented.
The current setup helps make sure `solang-aqd` works well with Polkadot, handling interactions with Polkadot and providing accurate responses to user requests.

Reply:
> Currently, to test solang-aqd, I'm using ink!-compiled contracts with the most recent version of substrate-contracts-node

> The current setup helps make sure solang-aqd works well with Polkadot, handling interactions with Polkadot and providing accurate responses to user requests.

This is obviously flawed. After all, `solang-aqd` is supposed to be tooling for solang and not `ink!`. The current setup does _not_ make sure that `solang-aqd` works well with Polkadot; it only makes sure that solang-aqd works with some random pre-compiled `ink!` contract. Anyone trying to use `solang-aqd` will just find out that it's broken (despite the CI being green), which is obviously flawed.

I mean testing against an `ink!` contract is fine if your goal is to be compatible with `ink!` contracts too. But it's not the main goal here.

> I'm using this approach until solang-substrate-ci gets updated.

This doesn't have anything to do with the node (if it had, obviously the solution would be to just use a compatible one on the CI). The contracts built by solang are fine and they run fine on all recent versions of pallet contracts.

As you already said. The problem is that `aqd` doesn't understand the metadata format. The real problem is that Solang contract do not include the [staticBufferSize](https://docs.rs/ink_metadata/5.0.0-alpha/ink_metadata/struct.EnvironmentSpecBuilder.html#method.static_buffer_size) field in the metadata. Which is fine because this comes from an unreleased version of `ink_metadata` (Solang uses the latest released version which is `4.3.0` and doesn't have this field yet).

If you encounter this problem, it's the wrong approach to just pretend everything is fine. The right way to do this is to state that the tooling does not yet work and to work on fixing this for good. This means you try to upgrade Solang to `ink-5.0.0-alpha` to make it work.

How this CI should be set up is to download a version of solang that is supposed to be compatible with your tool, then compile the contract on the CI and then use that to test everything.

We can merge this PR as it is, however
* This needs to be addressed afterwards
* The README should clearly state that the tool does not yet work for polkadot